core/remote/watcher.js

```diff
@@ -285,15 +285,18 @@ class RemoteWatcher {
             // move inside move
             change.was.path = remoteChange.applyMoveToPath(previousChange, change.was.path)
             change.needRefetch = true
-            return change
+            return change // FileMove
           } else {
-            return {
+            const descendantChange = {
               sideName,
-              type: 'IgnoredChange',
+              type: 'DescendantChange',
+              update: change.update,
               doc,
               was,
               detail: `File was moved as descendant of ${_.get(previousChange, 'doc.path')}`
             }
+            previousChange.type === 'DirMove' && remoteChange.includeDescendant(previousChange, descendantChange)
```

Comment:
Isn't `previousChange.type === 'DirMove'` already checked line 283?
Otherwise, detail above wouldn't make sense.
(I would assume it does i.e. this check is not needed)

Reply:
Flow is :sparkles: